Toolbar async prototype

nces opened the pull request.

nces committed:
Toolbar async prototype

nces commented:
Possible toolbar usage with async wrappers.

nces commented on Toolbar1Async/manifest.json:
```diff
@@ -0,0 +1,18 @@
+{
```
A couple things to note:
- The `PluginEntry` defines the "entry point" of the plugin, effectively the "index.html" for any website.
- There is no `toolbar.json`. This additional file seems unnecessary to me. If it's really desirable, I can make it work for the async case.
- Currently you can't make a panel and a toolbar in a single plugin, but that is completely doable, I just didn't workout what the manifest would look like in that case. e.g., it seems reasonable to be able to have a single "Plugin" that creates several panels and toolbar buttons, all driven off the same HTML page.
- It seems like in pretty much every use case, paths referred to in the manifest will refer to files relative to the plugin directory, which makes the use of `PLUGINLOCATION` superfluous, for relative paths at least. See comments in the corresponding FormIt360 PR.

jhauswirth replied:
I think being able to handle a plugin that has both a panel and toolbar would be good to have.  How much work would that be?

nces replied:
Looks like it already works (more or less).

josh-goldstein-adsk replied:
It's nice that a single plugin can be a panel and a toolbar. Nice work!

nces commented:
As with the original async wrapper change, this does not effect existing plugins.

The main motivation here (for me at least) is the single execution context for the user (eliminating the need for the `Scripts` manifest entry). See the comment in Toolbar1Async/manifest.json for some additional changes I made (because they were bothering me...).

jhauswirth commented on Toolbar1Async/index.html:
```diff
@@ -0,0 +1,22 @@
+            const pt1 = WSM.Geom.Point3d(-6, -6, 0);
+            const pt2 = WSM.Geom.Point3d(6, 6, 12);
+            const history = FormIt.GroupEdit.GetEditingHistoryID();
+            WSM.APICreateBlock(await history, await pt1, await pt2);
```
For each toolbar button you used to be able to assign a shortcut to the action.  How do you do that now?

nces replied:
Good question. That may not be possible. How did that work before?

jhauswirth replied:
FormIt.Commands.RegisterJSCommand("ToolbarPlugin.Button");
And in the json for the toolbar-
https://github.com/FormIt3D/FormItExamplePlugins/blob/master/Toolbar1/toolbar.json#L6

sum1991 reviewed (approved):
LGTM

nces committed:
Make async toolbar example also display a panel button

nces committed:
Rename Toolbar1Async -> ToolbarPanelAsync

nces commented:
I tried making some minor tweaks to the manifest and it looks like toolbars and panels can pretty easily coexist in a single plugin. I changed the example I added to showcase using shared code for toolbar buttons and a panel.
